roles/openshift_health_checker/openshift_checks/efk_stack_check.py

```diff
@@ -0,0 +1,277 @@
+#!/usr/bin/env python
+'''
+  Module for performing checks on an elasticseatch, fluentd, and kibana stack
+'''
+# pylint: disable=missing-docstring
+# vim: expandtab:tabstop=4:shiftwidth=4
+#
+#   Copyright 2017 Red Hat Inc.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# This is not a module, but pylint thinks it is.  This is a command.
+# pylint: disable=invalid-name
+# If a check throws an exception it is failed and should alert
+# pylint: disable=bare-except
+import ssl
+import urllib2
+import json
+
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class EFKStackCheck(OpenShiftCheck):
+    """
+        This is a check for the entire EFK stack shipped with OCP
+        https://github.com/openshift/openshift-tools/blob/prod/scripts/monitoring/cron-send-logging-checks.py
+    """
+
+    name = "efk_stack_check"
+    tags = ["elasticsearch", "efkstack"]
+
+    namespace = "logging"
+
+    es_pods = []
+    fluentd_pods = []
+
+    def __init__(self, module_executor):
+        super(EFKStackCheck, self).__init__(module_executor)
+
+        es_cert_prefix = '/etc/elasticsearch/secret/admin-'
+        self.es_curl = "curl -s --cert {}cert --key {}key --cacert {}ca -XGET ".format(es_cert_prefix, es_cert_prefix, es_cert_prefix)
+
+    def run(self, tmp, task_vars):
+        self.get_pods(task_vars)
+
+        status = dict()
+        status['elasticsearch'] = self.check_elasticsearch(task_vars)
+        status['fluentd'] = self.check_fluentd(task_vars)
+        status['kibana'] = self.check_kibana(task_vars)
+
+        if not status.get("elasticsearch", {}).get("pods", None):
+            return {"failed": True, "msg": "Unable to find any elasticsearch pods. Is elasticsearch installed on the host?"}
+
+        if not status.get("kibana"):
+            return {"failed": True, "msg": "Unable to access the Kibana logging url. No routes found."}
+        elif not status.get("kibana", {}).get("site_up"):
+            return {"failed": True, "msg": "Unable to access Kibana logging url. Invalid server response."}
+
+        return {"changed": False, "msg": status}
+
+    def oc(self, method, method_args, task_vars):
+        config_base = get_var(task_vars, "openshift", "common", "config_base")
+
+        args = {
+            "namespace": self.namespace,
+            "config_file": "".join([config_base, "/master/admin.kubeconfig"]),
```

Comment:
In this case we do not need `str.join`, we could simply write:

```python
config_base + "/master/admin.kubeconfig"
```

If we are to join parts, then we should at least go ahead and do not hard code the path separator...

```python
os.path.join(config_base, "master", "admin.kubeconfig")
```

(the effectiveness of this of course depends on how `config_base` itself is computed)